daemon/media_socket.c

```diff
@@ -1172,8 +1193,16 @@ static int stream_packet(struct stream_fd *sfd, str *s, const endpoint_t *fsin,
 
 	/* return values are: 0 = forward packet, -1 = error/dont forward,
 	 * 1 = forward and push update to redis */
-	if (rwf_in)
+	if (rwf_in) {
 		handler_ret = rwf_in(s, in_srtp);
+	}
+
+	// If recording pcap dumper is set, then we record the call.
+	if (recording_pdumper != NULL && call->record_call) {
+		stream_pcap_dump(recording_pdumper, stream, s);
+		call->recording->packet_num++;
```

Comment:
This seems to be insufficiently protected by a lock. Only stream-scope locks are held at this point, but it's modifying a call-scope variable. Am I right to assume that this counter is supposed to correspond to the packet index in the pcap? I don't know how thread-safe libpcap is and whether pcap_dump can be called without locking, as it seems to have the same problem. Either way, if the packet_num is supposed to correlate with the pcap packet index, then both have to be protected by the same lock, otherwise you may get out of order packets or mismatched indexes.